core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -117,11 +123,27 @@ private[spark] class NettyBlockTransferService(
         }
       }
```

Comment:
I think this patch is overcomplicated. We just need to change here
```
val blockFetchStarter = new RetryingBlockFetcher.BlockFetchStarter {
  override def createAndStart(blockIds: Array[String], listener: BlockFetchingListener) {
    val client = clientFactory.createClient(host, port)
    try {
      new OneForOneBlockFetcher(client, appId, execId, blockIds, listener,
        transportConf, tempFileManager).start()
    } catch (IOException e) {
      if (driverEndPointRef.askSync[Boolean](IsExecutorAlive(execId))) {
        throw e
      } else {
        new ExecutorDeadException
      }
    }
  }
}
```

Reply:
Amazing, I  will modify the code, thanks.

Reply:
@cloud-fan  I have made code refactoring, with extremely admire. Thanks!